File: Week3/Chaeyeon AN/day4-2.js
// https://school.programmers.co.kr/learn/courses/30/lessons/42586?language=javascript#

function solution(progresses, speeds) {
  var answer = [];
  let tempArr = [];

  for (let i = 0; i < speeds.length; i++) {
    progresses[i] = Math.ceil((100 - progresses[i]) / speeds[i]);
  }

  for (let i = 0; i < progresses.length; i++) {
    if (tempArr.length == 0 || progresses[i] <= tempArr[0]) {
      tempArr.push(progresses[i]);
    } else {
      answer.push(tempArr.length);
      tempArr = [];
      tempArr.push(progresses[i]);
    }

    if (i == progresses.length - 1) {
      answer.push(tempArr.length);
    }
  }

  return answer;
}

File: Week3/Chaeyeon AN/day4-2.py
# https://school.programmers.co.kr/learn/courses/30/lessons/42586?language=python3

import math

def solution(progresses, speeds):
    answer = []
    temp=[]
    
    for i in range (len(speeds)):
        progresses[i]=math.ceil((100-progresses[i])/speeds[i])
    
    for i in range (len(progresses)):
        if(len(temp)== 0 or progresses[i] <= temp[0] ):
            temp.append(progresses[i])
        else:
            answer.append(len(temp))
            temp=[]
            temp.append(progresses[i])
        if(i == len(progresses)-1):
            answer.append(len(temp))
    
    return answer